main.py

```diff
@@ -1,70 +1,93 @@
-    keep_swimming(name, friend_name, friend_relationship, friend_gender)
-
-def keep_swimming(name,friend_name,friend_relationship,friend_gender):
-    friends_length= input("How long did you know "+friend_gender+"? | ")
-    if any(ele in friends_length for ele in tps) is False:
-        friends_length: str=input(friends_length+"? What's that? Try giving me a more common time period. | ")
-    if any(ele in friends_length for ele in tps) is False:
-        print("Try again. Try using a common span of time. Like 1 year or something.")
-        keep_swimming(name,friend_name,friend_relationship,friend_gender)
-    animal: str=input("Cute. "+friends_length+". got it. That's not very long for a robot, but I respect that. If you could be any animal what would you be? | ")
-    if "a " in animal:
-        animal = animal.replace('a ',"")
-    animal_friend: str=input("Do you think that "+friend_name.title()+" would be the same animal? | ")
-    if animal_friend.lower().startswith('y'):
+
+    return name, friend_name, friend_relationship, friend_gender
+
+
+def keep_swimming(friend_name, friend_gender):
+    while True:
+        friends_length = input("How long did you know " + friend_gender + "? | ")
+        if valid_time(friends_length):
+            break
+        else:
+            print('What\'s that? Try giving me a known time unit. Years, Months, Days, etc. ')
+    friends_length = strip_identifiers(friends_length)
+
+    animal: str = input(
+        "Cute. " + friends_length + ". got it. That's not very long for a robot, but I respect that. "
+                                    "If you could be any animal what would you be? | ")
+    animal = strip_identifiers(animal)
+
+    while True:
+        animal_friend: str = input("Do you think that " + friend_name.title() + " would be the same animal?  (y/n) | ")
+        if animal_friend in ['y', 'n']:
+            break
+        else:
+            print('please only respond with \'y\' or \'n\'')
+    if animal_friend == 'y':
         animal_friend = animal
         print("Cute, you guys have so much in common.")
     else:
         print("lol. You guys aren't really friends if you wouldn't even be in the same animal family.")
-        animal_friend: str = input("What would they be? | ") ## need to incorporate gender better here
+        animal_friend: str = input("What would they be? | ")  ## need to incorporate gender better here
+
+    return friends_length, animal_friend, animal
 
-    think(name,friend_name,friend_relationship,friend_gender,friends_length,animal_friend,animal)
 
-def think(name,friend_name,friend_relationship,friend_gender,friends_length,animal_friend,animal):
+def think(name, friend_name, friend_relationship, friend_gender, friends_length, animal_friend, animal):
     print("Ok... let me think...")
-    time.sleep(1)
-    print("Thinking...")
-    time.sleep(1)
-    print("Thinking...")
-    time.sleep(1)
-    print("Thinking...")
+    for _ in range(3):
```

Comment:
Anytime you're doing the same thing over and over again, use a loop!

Reply:
in this case `for _ in range(3)` means do something three times. `_` is usually the number returned by range, but as I don't need the index, I use the `_` identifier to tell the interpreter 'yes I know this thing should have a variable here, but I don't need it, so I am intentionally ignoring it"